### PR TITLE
Add Spring Security Plugin

### PR DESCRIPTION
**Pull request by jojocodeX**
Add Spring Security Plugin

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/config/SecureConfig.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dubbo.config;
+
+public class SecureConfig extends AbstractConfig{
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/config/SecureConfig.java:20]: What is the purpose of defining this class?

File: dubbo-plugin/dubbo-spring-security/src/main/java/org/apache/dubbo/spring/security/filter/ContextHolderParametersSelectedTransferFilter.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import static org.apache.dubbo.spring.security.utils.SecurityNames.SECURITY_CONTEXT_HOLDER_CLASS_NAME;
+
+@Activate(group = CommonConstants.CONSUMER, order = -1)
+public class ContextHolderParametersSelectedTransferFilter
+    extends AndFilterConditionMatcher implements ConditionFilter, ClusterFilter {
+
+    public static List<FilterConditionMatcher> conditionMatchers = new ArrayList<>();
+
+    static{
+        conditionMatchers.add(new FilterConditionMatcherOnClass(SECURITY_CONTEXT_HOLDER_CLASS_NAME){
```
Comment (AlbumenJ) [inline dubbo-plugin/dubbo-spring-security/src/main/java/org/apache/dubbo/spring/security/filter/ContextHolderParametersSelectedTransferFilter.java:47]: Prevent use `static` in Dubbo for multi instance status purpose.

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-plugin/dubbo-spring-security/src/main/java/org/apache/dubbo/spring/security/filter/ContextHolderParametersSelectedTransferFilter.java (hunk 2 above)
Comment (jojocodeX) [reply dubbo-plugin/dubbo-spring-security/src/main/java/org/apache/dubbo/spring/security/filter/ContextHolderParametersSelectedTransferFilter.java:48]: 检查在加载时就已经决定了，没有必要实例化过程中检查

File: dubbo-common/src/main/java/org/apache/dubbo/config/SecureConfig.java (hunk 1 above)
Comment (jojocodeX) [reply dubbo-common/src/main/java/org/apache/dubbo/config/SecureConfig.java]: 到时考虑到一些安全的配置，现阶段暂时不需要，已经移出

File: dubbo-plugin/dubbo-spring-security/src/main/java/org/apache/dubbo/spring/security/utils/ObjectMapperCodec.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+final public class ObjectMapperCodec {
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    static{
+        mapper.registerModule(new CoreJackson2Module());
+    }
+
+    private ObjectMapperCodec(){}
```
Comment (AlbumenJ) [inline dubbo-plugin/dubbo-spring-security/src/main/java/org/apache/dubbo/spring/security/utils/ObjectMapperCodec.java:32]:
```suggestion
    private final ObjectMapper mapper = new ObjectMapper();

    public ObjectMapperCodec(){
        mapper.registerModule(new CoreJackson2Module());
    }
```

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-plugin/dubbo-spring-security/src/main/java/org/apache/dubbo/spring/security/utils/ObjectMapperCodec.java (hunk 3 above)
Comment (jojocodeX) [reply dubbo-plugin/dubbo-spring-security/src/main/java/org/apache/dubbo/spring/security/utils/ObjectMapperCodec.java]: ok

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**